debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlSchema.java

```diff
@@ -298,45 +301,52 @@ public boolean applyDdl(SourceInfo source, String databaseName, String ddlStatem
                 throw e;
             }
         } finally {
-            if (statementConsumer != null) {
-
-                // We are supposed to _also_ record the schema changes as SourceRecords, but these need to be filtered
-                // by database. Unfortunately, the databaseName on the event might not be the same database as that
-                // being modified by the DDL statements (since the DDL statements can have fully-qualified names).
-                // Therefore, we have to look at each statement to figure out which database it applies and then
-                // record the DDL statements (still in the same order) to those databases.
-
-                if (!ddlChanges.isEmpty() && ddlChanges.applyToMoreDatabasesThan(databaseName)) {
-
-                    // We understood at least some of the DDL statements and can figure out to which database they apply.
-                    // They also apply to more databases than 'databaseName', so we need to apply the DDL statements in
-                    // the same order they were read for each _affected_ database, grouped together if multiple apply
-                    // to the same _affected_ database...
-                    ddlChanges.groupStatementStringsByDatabase((dbName, ddl) -> {
-                        if (filters.databaseFilter().test(dbName) || dbName == null || "".equals(dbName)) {
-                            if (dbName == null) dbName = "";
-                            statementConsumer.consume(dbName, ddlStatements);
-                        }
-                    });
-                } else if (filters.databaseFilter().test(databaseName) || databaseName == null || "".equals(databaseName)) {
-                    if (databaseName == null) databaseName = "";
-                    statementConsumer.consume(databaseName, ddlStatements);
+            changes = tables.drainChanges();
+            // No need to send schema events or store DDL if no table has changed
+            if (!storeOnlyMonitoredTablesDdl || !changes.isEmpty()) {
```

Comment:
Why is it that you don't apply the table filter here (unlike below)? IIUC this would still emit non-whitelisted table changes on the (public) schema changes topic? Is that on purpose?

I'd say the new option should control consistently whether changes will be included in the public schema change topic _and_ in the internal history topic.

Reply:
Weel, I made this intentional - the reason is that I see this as optimization of our internal strctures but I tgout it is best for user to get all availble information in the public topic so it can do filtering on its own. Also the public topic is not prone to limitation in compaction/deletes etc. so I have not felt a pressure to remove it from there,

Reply:
Ok, that's convincing :) In that case, can you just add some comment then as suggested below.